logstash-core/lib/logstash/util/dead_letter_queue_factory.rb

```diff
@@ -0,0 +1,32 @@
+require 'logstash/environment'
+
+module LogStash; module Util
+  class DeadLetterQueueWriter
+
+    def initialize(inner_writer)
+      @inner_writer = inner_writer
+    end
+
+    def write(logstash_event, plugin_type, plugin_id, reason)
```

Comment:
not unless we inject it into the plugin instance somehow without the user knowing?

e.g. new constructor or setter